heon/generate_flow_seq.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+import random
+
+
+def flush_file(file):
+    f = open(file, "w")
+    f.write("")
+    f.close()
+
+
+def write_flow_num_to_file(file, value):
+    f = open(file, "a")
+    f.write(f"{value}\n")
+    f.close()
+
+
+def run_single_iteration(flow_num_list, file_to_write):
+    list_to_choose_from = flow_num_list.copy()
+    for i in range(len(list_to_choose_from)):
+        chosen_num = random.choice(list_to_choose_from)
+        write_flow_num_to_file(file_to_write, chosen_num)
+        list_to_choose_from.remove(chosen_num)
+
+
+def generate_random_flow_sequence_grouped(flow_num_list, num_iter, file_to_write):  
+    for i in range(0, num_iter):
+        run_single_iteration(flow_num_list, file_to_write)
+        
+
+def generate_random_flow_sequence_independent(num_conn, flow_num_list, num_iter, file_to_write):
+    flow_num_remaining_count_dict = {}
+    for i in range(1, num_conn+1):
+        flow_num_remaining_count_dict[i] = num_iter
+        
+    list_to_choose_from = flow_num_list.copy()
+    for i in range(0, num_conn * num_iter):
+        chosen_num = random.choice(list_to_choose_from)
+        write_flow_num_to_file(file_to_write, chosen_num)
+        remaining_count = flow_num_remaining_count_dict[chosen_num]
+        remaining_count = remaining_count - 1
+        flow_num_remaining_count_dict[chosen_num] = remaining_count
+        if remaining_count <= 0:
+            list_to_choose_from.remove(chosen_num)
+
+
+def generate_desired_flow_sequence(flow_num_list, num_iter, file_to_write):
+    for i in range(0, num_iter):
+        for flow_num in flow_num_list:
+            write_flow_num_to_file(file_to_write, flow_num)
+    
+
+def validate_args(args):
+    if args.num_conn <= 0: 
+        raise ValueError
+    if args.iterations <= 0:
+        raise ValueError
+
+
+if __name__ == "__main__":
+    # Parse arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-n", "--num-conn", type=int, required=False,
+                        default=3, help="Number of client connections")
+    parser.add_argument("-l", "--iterations", type=int, required=False, default=40,
+                        help="Number of iterations. The length of the flow sequence is equal to num_conn * iterations. (ex: 3 * 40 = 120 lines)")
+    parser.add_argument("-r", "--random-flow-seq-file", type=str,
+                        required=False, default="random_flow_sequence.txt", help="Name of file to store randomly generated flow sequence")
+    parser.add_argument("-d", "--desired-flow-seq-file", type=str,
+                        required=False, default="desired_flow_sequence.txt", help="Name of file to store desired flow sequence")
+    parser.add_argument("-c", "--chosen-seq-type", type=str, choices=["desired", "random"], default="desired", help="Type of flow sequence to store to the file that contains input flow sequence for the client side")
```

Comment:
I didn't completely follow the reason for this argument's existence. 
Won't we need to generate the input and desired in each case, why generate them separately?

Reply:
I may be wrong, but I thought it would be helpful to have control over what the client gets as the input sequence. By having this option, we can test the perfectly ideal input (that resembles the desired output). This is helpful because non-coflow qdiscs cannot output the desired flow sequence unless the input flow sequence itself was in the desired order (assuming no reordering).

Reply:
(different kinds of randomness)